consultation_analyser/consultations/views/answers.py

```diff
@@ -50,3 +52,93 @@ def index(
     }
 
     return render(request, "consultations/answers/index.html", context)
+
+
+@login_required  # TODO: check if we still need this
+@user_can_see_consultation
+def show(
+    request: HttpRequest,
+    consultation_slug: str,
+    question_slug: str,
+    response_id: int,
+):
+    consultation = get_object_or_404(models.Consultation, slug=consultation_slug)
+    question = get_object_or_404(models.Question, slug=question_slug, consultation=consultation)
+    response = get_object_or_404(models.Answer, id=response_id)
+
+    # TODO: update with QuestionPart
+    all_theme_mappings = models.ThemeMapping.get_latest_theme_mappings_for_question_part(
+        part=response.question_part
+    )
+    all_themes = models.Theme.objects.filter(id__in=all_theme_mappings.values("theme"))
+    existing_themes = models.ThemeMapping.objects.filter(answer=response).values_list(
+        "theme", flat=True
+    )
+
+    if request.method == "POST":
+        requested_themes = request.POST.getlist("theme")
+        existing_mappings = models.ThemeMapping.objects.filter(answer=response)
+
+        # themes to delete
+        existing_mappings.exclude(theme_id__in=requested_themes).delete()
+
+        # themes to add
+        themes_to_add = set(requested_themes).difference([str(theme) for theme in existing_themes])
+        for theme_id in themes_to_add:
+            models.ThemeMapping.objects.create(
```

Comment:
`reason` is one of the fields that's an output of the AI pipeline, not sure if used any more. 
We should be able to determine that this is a user generated theme mapping from the history table (maybe something to check in tests?).